destination/destination.go

```diff
@@ -59,29 +63,47 @@ func (d *Destination) Configure(ctx context.Context, cfg config.Config) error {
 		return fmt.Errorf("failed to parse destination config : %w", err)
 	}
 
+	switch queue := d.config.QueueName; {
+	case isGoTemplate(queue):
+		parser, err := parserFromGoTemplate(queue)
+		if err != nil {
+			return fmt.Errorf("failed to create template parser: %w", err)
+		}
+		d.parseQueueName = parser
+	case queue != "" && !d.config.UseQueueName:
+		d.parseQueueName = parserFromCollectionWithDefault(queue)
+	case queue != "" && d.config.UseQueueName:
+		d.parseQueueName = staticParser(queue)
+	default:
+		d.parseQueueName = parseAlwaysFromCollection
+	}
+
 	return nil
 }
 
 func (d *Destination) Open(ctx context.Context) (err error) {
 	d.svc, err = common.NewSQSClient(ctx, d.httpClient, d.config.Config)
-
 	if err != nil {
 		return fmt.Errorf("failed to create destination sqs client: %w", err)
 	}
 
-	queueInput := &sqs.GetQueueUrlInput{
-		QueueName: &d.config.AWSQueue,
-	}
+	d.queueURLMap = newQueueURLMap(d.svc)
 
-	// Get URL of queue
-	urlResult, err := d.svc.GetQueueUrl(ctx, queueInput)
-	if err != nil {
-		return fmt.Errorf("failed to get sqs queue url : %w", err)
-	}
+	if !isGoTemplate(d.config.QueueName) {
+		// we can be nice to the user and inform them that the queue that they did
+		// specify exists and that the connector has access to it.
```

Comment:
is this comment a todo? we can address this and delete the comment if so

Reply:
It isn't, but I agree that the wording isn't really clear. Let me fix that.